numpy/lib/_iotools.py

```diff
@@ -832,25 +846,29 @@ def update(self, func, default=None, testing_value=None,
         """
         self.func = func
         self._locked = locked
+
         # Don't reset the default to None if we can avoid it
         if default is not None:
             self.default = default
             self.type = self._dtypeortype(self._getdtype(default))
         else:
             try:
-                tester = func(testing_value or b'1')
+                tester = func(testing_value or '1')
             except (TypeError, ValueError):
                 tester = None
             self.type = self._dtypeortype(self._getdtype(tester))
-        # Add the missing values to the existing set
-        if missing_values is not None:
-            if _is_bytes_like(missing_values):
-                self.missing_values.add(missing_values)
-            elif hasattr(missing_values, '__iter__'):
-                for val in missing_values:
-                    self.missing_values.add(val)
+
+        # Add the missing values to the existing set or clear it.
+        if missing_values is None:
+            # Clear all missing values even though the ctor initializes it to
+            # {''} when the argument is None.
+            self.missing_values = {}
```

Comment:
This is a bug - it was presumably supposed to use `set()`, but ends up producing a `dict` instead.

Does this code path ever get taken?

Reply:
I don't think so, certainly not tested, as previously used `append` for a `set`.

Reply:
I thought it used `update`, which is defined on both?

Maybe we get away with it because later on we test it for truthiness.

Reply:
I think originally it was all written using a list, then someone upgraded to sets and missed fixing up this bit of code.

Reply:
Nope, used `append` in a loop.